### PR TITLE
[CI] Github Action trigger condition optimized for cost saving

### PR DESCRIPTION
**Pull request by nwangfw**
## Pull Request Description
Github Action trigger condition is optimized. Now each test will only be triggered when the corresponding code is modified.

## Related Issues
Resolves: https://github.com/aibrix/aibrix/issues/253

**Important: Before submitting, please complete the description above and review the checklist below.**

### CODE REVIEW

File: .github/workflows/release-build.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: Release Build and Push
 
 on:
+  workflow_dispatch:  # Allows manual trigger
```
Comment (Jeffwan) [inline .github/workflows/release-build.yaml:4]: this makes sense

File: .github/workflows/python-tests.yml [hunk 2]
```diff
@@ -1,11 +1,15 @@
 name: Python Tests
 
 on:
+  workflow_dispatch:  # Allows manual trigger
```
Comment (Jeffwan) [inline .github/workflows/python-tests.yml]: do we want manual trigger here?

File: .github/workflows/docker-push-images.yml [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+      - 'Dockerfile'
+      - 'build/**'
+      - 'pkg/**'
+      - 'cmd/**'
+      - 'python/**'
+      - 'Makefile'
+      - 'go.mod'
+      - 'go.sum'
+    tags:
+      - 'v*'
```
Comment (Jeffwan) [inline .github/workflows/docker-push-images.yml]: we don't push to tags (they are protected). This is something unnecessary I think.

File: .github/workflows/python-tests.yml (hunk 2 above)
Comment (nwangfw) [reply .github/workflows/python-tests.yml]: deleted

File: .github/workflows/docker-push-images.yml (hunk 3 above)
Comment (nwangfw) [reply .github/workflows/docker-push-images.yml]: deleted

**Pull request merged**